rusk-wallet/src/bin/main.rs

```diff
@@ -85,12 +86,44 @@ where
 
     // check for connection errors
     match con {
-        Err(Error::RocksDB(e)) => panic!{"Please reset the cache! {e}"},
-        Err(e) => warn!("[OFFLINE MODE]: Unable to connect to Rusk, limited functionality available: {e}"),
+        Err(Error::RocksDB(e)) => {
+            wallet.close();
+
+            let msg = match e.kind() {
+                ErrorKind::InvalidArgument => {
+                    format!("You've used the wrong key to open the cache database \n\r\n\r{0: <1} delete the cache? (Alternatively specify -p (--profile) flag to open with a different cache db)", "[ALERT]")
+                },
+                ErrorKind::Corruption => {
+                       format!("The database appears to be corrupted \n\r\n\r{0: <1} delete the cache?", "[ALERT]")
+                },
+                _ => {
+                    format!("Unknown database error {:?} \n\r\n\r{1: <1} delete the cache?", e, "[ALERT]")
+                }
+            };
+
+             match prompt::ask_confirm_erase_cache(&msg)? {
+                true => {
+                    if let Some(io_err) = wallet.delete_cache().err() {
+                        error!("Error while deleting the cache: {io_err}");
+                    }
+
+                    info!("Restart the wallet to create new cache and sync with network");
+                },
+                false => {
+                    info!("Wallet will now exit, reset the cache manually");
```

Comment:
If the user selected `no` when asked to delete the cache, they might also not want to delete it manually so adding `, reset the cache manually` is not really needed. I think here you can merely let the user know that the wallet cannot proceed and will now exit.